### PR TITLE
compile the service processDocuments

### PR DESCRIPTION
**Pull request by jianmingtu**

### CODE REVIEW

File: jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/models/GuidMapDocument.java [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
 @AllArgsConstructor
-public class GuidMapDocument implements Serializable {
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "GuidMapDocument")
+public class GuidMapDocument {
     private String version;
-    private Map<String, String> mapping;
+
+    @XmlElement(name = "Mapping")
+    private List<GuidDocumentMapping> Mapping;
```
Comment (sylar-zeyangz) [inline jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/models/GuidMapDocument.java]: please follow the conventional naming standard

File: jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/models/GuidMapDocument.java [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
 
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
-public class GuidMapDocument implements Serializable {
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "GuidMapDocument")
+public class GuidMapDocument {
     private String version;
-    private Map<String, String> mapping;
```
Comment (sylar-zeyangz) [inline jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/models/GuidMapDocument.java:14]: is there a reason we have to use self-defined map structure?

**Review by sylar-zeyangz — CHANGES_REQUESTED** [review]

File: jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/controllers/TransmitController.java [hunk 3]
```diff
@@ -56,10 +56,14 @@ public TransmitController(
         this.objectMapper = objectMapper;
     }
 
+    // Source from :    http://reeks.bcgov/
```
Comment (sylar-zeyangz) [inline jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/controllers/TransmitController.java]: source differs, might not be a good idea to add it here

**Review by sylar-zeyangz — CHANGES_REQUESTED** [review]

File: jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/models/GuidMapDocument.java (hunk 2 above)
Comment (jianmingtu) [reply jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/models/GuidMapDocument.java:14]: database side checks if API sends a valid xml file but previously we only sent the map

File: jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/controllers/TransmitController.java (hunk 3 above)
Comment (jianmingtu) [reply jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/controllers/TransmitController.java]: removed source

File: jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/models/GuidMapDocument.java (hunk 1 above)
Comment (jianmingtu) [reply jag-crdp-application/src/main/java/ca/bc/gov/open/crdp/models/GuidMapDocument.java]: changed

**Review by sylar-zeyangz — APPROVED** [review]

**Pull request merged**